src/test/java/org/dcsa/ebl/controller/TransportDocumentControllerTest.java

```diff
@@ -0,0 +1,174 @@
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@DisplayName("Tests for TransportDocumentController")
+@ActiveProfiles("test")
+@WebFluxTest(controllers = {TransportDocumentController.class})
+@Import(value = {GlobalExceptionHandler.class, SecurityConfig.class})
+class TransportDocumentControllerTest {
+
+  @Autowired WebTestClient webTestClient;
+
+  @MockBean TransportDocumentService transportDocumentService;
+
+  private final String TRANSPORT_DOCUMENT_ENDPOINT = "/transport-documents";
+
+  TransportDocumentTO transportDocumentTO;
+
+  @BeforeEach
+  private void init() {
+    transportDocumentTO = new TransportDocumentTO();
+    transportDocumentTO.setTransportDocumentReference("TransportDocumentReference1");
+
+    Address address = new Address();
+    address.setCity("Amsterdam");
+    address.setCountry("Netherlands");
+    address.setStreet("Strawinskylaan");
+    address.setPostalCode("1077ZX");
+    address.setStreetNumber("4117");
+    address.setFloor("6");
+    address.setStateRegion("Noord-Holland");
+
+    LocationTO locationTO = new LocationTO();
+    locationTO.setLocationName("DCSA Headquarters");
+    locationTO.setAddress(address);
+    locationTO.setId("1");
+
+    ChargeTO chargeTO = new ChargeTO();
+    chargeTO.setChargeTypeCode("chargeTypeCode");
+    chargeTO.setCalculationBasis("CalculationBasics");
+    chargeTO.setCurrencyAmount(100.0);
+    chargeTO.setCurrencyCode("EUR");
+    chargeTO.setQuantity(1.0);
+    chargeTO.setUnitPrice(100.0);
+    chargeTO.setPaymentTermCode(PaymentTerm.COL);
+
+    CarrierClauseTO carrierClauseTO = new CarrierClauseTO();
+    carrierClauseTO.setClauseContent("CarrierClause");
+
+    ShippingInstructionTO shippingInstructionTO = new ShippingInstructionTO();
+    shippingInstructionTO.setIsShippedOnboardType(true);
+    shippingInstructionTO.setIsElectronic(true);
+    shippingInstructionTO.setIsToOrder(true);
+    shippingInstructionTO.setShippingInstructionID(UUID.randomUUID().toString());
+    shippingInstructionTO.setDocumentStatus(ShipmentEventTypeCode.RECE);
+    shippingInstructionTO.setPlaceOfIssueID(locationTO.getId());
+    shippingInstructionTO.setAreChargesDisplayedOnCopies(true);
+
+    ShipmentTO shipmentTO = new ShipmentTO();
+    shipmentTO.setBooking(new BookingTO());
+    shippingInstructionTO.setShipments(List.of(shipmentTO));
+
+    transportDocumentTO = new TransportDocumentTO();
```

Comment:
Duplicate Initialization?

Reply:
fixed, duplicate initialization is removed